x/ibc/04-channel/keeper/keeper.go

```diff
@@ -230,6 +230,26 @@ func (k Keeper) GetAllPacketCommitments(ctx sdk.Context) (commitments []types.Pa
 	return commitments
 }
 
+// IteratePacketCommitmentAtChannel provides an iterator over all PacketCommmitment objects
+// at a specified channel. For each packet commitment, cb will be called. If the cb returns
+// true, the iterator will close and stop.
+func (k Keeper) IteratePacketCommitmentAtChannel(ctx sdk.Context, portID, channelID string, cb func(_, _ string, sequence uint64, hash []byte) bool) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte(host.KeyPacketCommitmentPrefix+host.ChannelPath(portID, channelID)))
+	k.iterateHashes(ctx, iterator, cb)
+}
+
+// GetAllPacketCommitmentsAtChannel returns all stored PacketCommitments objects for a specified
+// port ID and channel ID.
+func (k Keeper) GetAllPacketCommitmentsAtChannel(ctx sdk.Context, portID, channelID string) (commitments []types.PacketAckCommitment) {
+	k.IteratePacketCommitmentAtChannel(ctx, portID, channelID, func(_, _ string, sequence uint64, hash []byte) bool {
```

Comment:
is there a better approach here other than doing `_, _ string` without duplicating the iterator code too much?